### PR TITLE
Social: Added Bluesky to social previews

### PR DESCRIPTION
**Pull request by manzoorwanijk**
## Proposed changes:

* Social: Added Bluesky to social previews

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Ensure that your blog has the `jetpack-social-bluesky-connection` blog sticker
* Run `pnpm install` and `jetpack build plugins/jetpack plugins/social`
* Goto post editor
* Add a Bluesky connection
* Click on "Preview social posts"
* Confirm that you see the Bluesky preview
* Confirm that it matches the previews on Bluesky with and without custom media
* Click on "Preview" in Social Previews panel
image

* Goto Bluesky preview
* Confirm that you see the Bluesky preview for Your post and Link preview sections
* Confirm that both look fine
* Now close the modal and remove the Bluesky connection
* Click on "Preview" again
* Confirm that you don't see the "Your post" section

<table>
    <tbody>
        <tr>
            <th>Has featured image</th>
            <th>Has custom media</th>
            <th>Result</th>
        </tr>
        <tr>
            <th colspan="3">Social post preview</th>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> Screenshot 2024-10-07 at 1 44 08 PM
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> Screenshot 2024-10-07 at 1 44 17 PM
            </td>
        </tr>
        <tr>
            <td> ❌ </td>
            <td> ❌ </td>
            <td> Screenshot 2024-10-07 at 2 09 47 PM
            </td>
        </tr>
        <tr>
            <th colspan="3">Common Social preview</th>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> Screenshot 2024-10-07 at 1 45 58 PM
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> Screenshot 2024-10-07 at 1 46 12 PM
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> Screenshot 2024-10-07 at 1 44 54 PM
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> Screenshot 2024-10-07 at 1 45 15 PM
            </td>
        </tr>
        <tr>
            <td> ❌ </td>
            <td> ❌ </td>
            <td> Screenshot 2024-10-07 at 2 10 05 PM
            </td>
        </tr>
    </tbody>
</table>

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/social-previews/mastodon.js [hunk 1]
```diff
@@ -40,7 +38,6 @@ const MastodonPreview = props => {
 			description={ content }
 			customText={ message }
 			customImage={ customImage }
-			isSocialPost={ attachedMedia.length > 0 }
```
Comment (gmjuhasz) [inline projects/js-packages/publicize-components/src/components/social-previews/mastodon.js:43]: Just for clarity, why are we removing this?

**Review by gmjuhasz — COMMENTED** [review]
I think when we show the link previews, we should not show the link as text before it. Now that I checked it, that's how it works for all other networks except Mastodon, it has to be fixed there, but that can be a follow-up.

That's how I implemented it in [my task](https://github.com/Automattic/jetpack-reach/issues/609) for the backend following Sid's description.

CleanShot 2024-10-07 at 12 04 15 png

CleanShot 2024-10-07 at 12 00 24 png

File: projects/js-packages/publicize-components/src/components/social-previews/mastodon.js (hunk 1 above)
Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/social-previews/mastodon.js:43]: It's not used anywhere.

**Comment by manzoorwanijk** [comment]
> I think when we show the link previews, we should not show the link as text before it.

But that is how we send it to Bluesky.

**Comment by gmjuhasz** [comment]
> But that is how we send it to Bluesky.

Yeah, but Facebook and most networks create the link preview from the link; for Bluesky we need to use [embeds](https://docs.bsky.app/docs/tutorials/creating-a-post#website-card-embeds) similarly to media, so on the backend we transform the link into an embed which I did here: D162910-code

This was the task description there
>  Try sharing a post to Facebook and see the link preview. We can replicate the same behaviour with Bluesky as well, i.e don't include the link in the body of the post, just show the excerpt and the link preview.

**Comment by manzoorwanijk** [comment]
I get it now. I tried to create previews for the current state of posts that are sent out. We can come back later when that link preview diff is merged.

**Comment by gmjuhasz** [comment]
Yeah, sounds good. that should get merged soon, Sid wanted to review it today, maybe tomorrow when he is back

**Comment by manzoorwanijk** [comment]
@gmjuhasz I have created a fix in Calypso package and published to npm - https://github.com/Automattic/wp-calypso/pull/95249

- When there is custom media attached

    Screenshot 2024-10-09 at 5 18 36 PM

- When there is no custom media

    Screenshot 2024-10-09 at 5 18 07 PM

**Review by gmjuhasz — DISMISSED** [review]
Tests well for me, nice one!

**Pull request merged**